docs/build-configuration/environment-configuration.md

```diff
@@ -0,0 +1,29 @@
+---
+layout: "docs"
+title: Environment Configuration
+class: documentation
+permalink: /build/environment-configuration/
+published: true
+---
+
+You have access to certain environment options while configuring your Probo builds. These options are set as top level keys, on the same level as `assets` or `steps` in your `probo.yaml` file.
+
+## Allow Environment Access While Building
+
+We assume environments have confidential data and can be in potentially insecure states while building. Because of this, our default settings make sure environments are not made public until all of the steps have completed and the build is finished.
```

Comment:
It makes is so that it can be accessed while still building. That may be from an external service or a human. We can't/don't differentiate.